File: frontend/src/app/core/models/asiento.ts
export enum EstadoAsiento {
  LIBRE = 'LIBRE',
  OCUPADO = 'OCUPADO',
  MANTENIMIENTO = 'MANTENIMIENTO'
}

export interface Asiento {
  id: number;
  numero: string;
  estado: EstadoAsiento;
  precio: number;
  salaId: number;
  horarioId: number;
}

File: catalogo/src/main/java/cinerama/catalogo/models/Asiento.java
package cinerama.catalogo.models;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name="Asientos")
@Data
public class Asiento {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "numero", nullable = false)
    private String numero; // Ej: "A1", "F12"

    @Enumerated(EnumType.STRING)
    @Column(name = "estado", nullable = false)
    private EstadoAsiento estado;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "horario_id", nullable = false)
    private Horario horario;
}
